tests/json_test.cpp

```diff
@@ -243,10 +245,9 @@ TEST_CASE( "translation_text_style_check", "[json][translation]" )
             R"(    Suggested fix: insert " ")" "\n"
             R"(    At the following position (marked with caret))" "\n"
             R"()" "\n"
-            R"("…foo.)" "\n"
-            R"(       ^)" "\n"
-            R"(         bar.")" "\n" ),
-        R"("…foo. bar.")" ); // NOLINT(cata-text-style)
+            R"("…foo. bar.")" "\n"
+            R"(      ▲▲▲)" "\n" ),
```

Comment:
This shows that when there are unicode characters in the message, the previous method allows seeing the exact position of the error by breaking the line.

Reply:
The previous method didn't work for unicode either as most json code appears to assume all-ascii-content - it'll happily break line in the middle of a codepoint, e.g. place `"type": 两` in a json file and try to run any formatter or execute the game you'll see broken text:

![image](https://github.com/CleverRaven/Cataclysm-DDA/assets/6560075/583f6e2d-f022-4412-ade5-1b064038b87c)

I think optimizing for the edge case which doesn't get fully solved isn't going to benefit the main target group (json writers) that much, instead we can optimize for the common case; json errors with mostly latin or non-"duo"spaced/cjk content and leave the outliers where they are now (complex errors are usually resolved in discord anyway)

Reply:
I think it is trying to read in the first character to determine the value type, so it assumes the character is ascii and breaks the line in the wrong place.

I don't quite follow your second paragraph though, but there are mod writers who writes text primarily in Chinese (and presumably other languages) so I think it's important that it handles unicode characters correctly.

Reply:
Yep it does, that's what i was trying to say in the second paragraph - instead of assuming the terminal/editor is broken and using line breaks as cursor (which performs worse than incorrectly placed cursor in case of problems) we could assume most users have their terminal correctly configured for their locale and will place the cursor at or very near the correct spot.

In current proposed code it does so when testing with some CJK, cyrillic and ukrainian symbols with pretty much 0 config, it's likely impossible to exactly match the width of every codepoint on every font but it should be close enough for most users

There's also the fallback option of using line/column numbers (some editors will even follow them automatically) or using the game to render which will make the width match exactly